### ISSUE TITLE
After deployment was renamed, old replica sets not removed

### ISSUE THREAD

**Issue by whitecolor** [helm/helm#1811]
I changed a template files of there deployed release and upgraded it. Changes affected deployment and service names, the result was:

- new deployment with new name created.
- new service with new name created
- old service was removed
- old deployment was removed
- new replica set (with new name) and pods where created
- old replica set (with old name) and pods **were NOT deleted**

**Comment by technosophos** [comment]
This is the replica set attached to the (old) deployment object? That sounds like a bug in Kubernetes.

You can look in the Tiller logs, where you should see something about a reaper being started for your old deployment. That is where Helm requests that Kubernetes clean up the deployment object.

What version of Kubernetes are you running?

**Comment by whitecolor** [comment]
I'm on GCE, see, two Replica sets managed by helm without pods:

![image](https://cloud.githubusercontent.com/assets/736697/21847715/1d3a7ada-d81f-11e6-8eee-f73f815fc1aa.png)

**Comment by technosophos** [comment]
Okay, I'm not totally positive about this (cc @lachie83 ), but I _think_ GCE is still on Kubernetes 1.4, which wasn't reaping replica sets that attached to deployments. I'm going to post this issue in the Slack to see if anyone else might be able to add more context, since I'm unsure.

**Comment by so0k** [comment]
via https://cloud.google.com/container-engine/release-notes

at this time of writing:
>Kubernetes v1.4.7 is the default version for new clusters.
>
>The following Kubernetes versions are available for new clusters:
>
>- 1.4.7
>- 1.5.1
> 
>To create a cluster at a version which is not the default, run:
>
>`gcloud container clusters create --cluster-version=$VERSION $NAME`
>The following Kubernetes versions are available for container cluster node upgrades/downgrades:
>
>- 1.5.1
>- 1.4.7
>- 1.4.6
>- 1.3.10
>- 1.2.7

**Comment by whitecolor** [comment]
Ok, I will try to update the Cluster version and report back.

**Comment by technosophos** [comment]
Any news on this one?

**Comment by technosophos** [comment]
Did Kube 1.5 solve this one?

**Comment by whitecolor** [comment]
I think it did. Closing.

**Comment by fbird** [comment]
But i got same issue with Kube 1.6  when i use helm upgrade my deployment

**Comment by patvdleer** [comment]
Same issues here, I'm running `1.6.11-gke.0` on gcloud, deployments are done via gitlab using helm.

**Comment by francisdb** [comment]
We see the same with `1.8.2-gke.0`
Should we create a new ticket?

**Comment by bacongobbler** [comment]
I'll re-open this one. Would you all please send us reports on how/what occured, and what you did (if anything) to resolve the issue?

**Comment by francisdb** [comment]
how we reproduce:

1. update image version in chart
2. `helm upgrade staging -f inventory/staging-environment.yaml ./foo-environment`
3. we see one extra replica set

foo-environment controls 3 pods with each one container
we changed version for one pod
as expected pod indicates one container `Terminating`, one `ContainerCreating`

Is there a way to get logs for the actions performed on the cluster?

**Comment by lkt82** [comment]
We are also running Kubernetes 1.9.6 and have this issue when we use the helm upgrade --install command

**Comment by dverbeek84** [comment]
Same here only i am using EKS

**Comment by gaieges** [comment]
@bacongobbler It looks like some others (and myself) are still seeing this issue. Can we reopen this ticket?

I'm using AKS on k8s `1.10.6` and helm `2.10.0` and basically see what everyone else is seeing

**Comment by florianrusch** [comment]
And another one: Everytime the deployment resource change, we are getting a new replicaset and all old ones are still existing.

Kubernetes Version: (on-prem)
```Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T10:39:04Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:36:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```
Helm Version:
```
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```

**Comment by PacoDu** [comment]
We face the same issue: replicasets are not removed after a helm upgrade (for harbor helm chart)

Kubernetes version:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

Helm version:
```
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.1", GitCommit:"5270352a09c7e8b6e8c9593002a73535276507c0", GitTreeState:"clean"}
```

**Comment by bacongobbler** [comment]
this should be fixed in Helm 3 with #6124. Please feel free to test out the alpha and let us know if you run into issues with this case. Thanks!

**Comment by bacongobbler** [comment]
see also https://github.com/helm/helm/issues/5915
